### PR TITLE
`Development`: Speedup client tests for exercise scores

### PR DESCRIPTION
**Pull request by ole-ve**
#### General
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).

### Motivation and Context
This PR speeds up the client tests in exercise-scores.component.spec.ts.

### Description
Changed usage of chai/sinon to jest, mocked services (including adding needed mock methods).

#### Code Review
- [ ] Review 1
- [ ] Review 2

### CODE REVIEW

File: src/test/javascript/spec/component/exercises/shared/exercise-scores.component.spec.ts [hunk 1]
```diff
@@ -108,12 +100,12 @@ describe('Exercise Scores Component', () => {
...
             ],
             providers: [
-                { provide: TranslateService, useClass: MockTranslateService },
                 { provide: ActivatedRoute, useValue: route },
-                { provide: AlertService, useClass: MockAlertService },
-                { provide: Router, useValue: router },
                 { provide: LocalStorageService, useClass: MockSyncStorage },
                 { provide: SessionStorageService, useClass: MockSyncStorage },
+                { provide: ExerciseService, useClass: MockExerciseService },
+                { provide: AccountService, useClass: MockAccountService },
```
Comment (b-fein) [inline src/test/javascript/spec/component/exercises/shared/exercise-scores.component.spec.ts:107]:
```suggestion
```
The `ArtemisTestModule` already provides the `AccountService`.

File: src/test/javascript/spec/component/exercises/shared/exercise-scores.component.spec.ts [hunk 2]
```diff
@@ -267,34 +260,35 @@ describe('Exercise Scores Component', () => {
...
+        const resultServiceStub = jest.spyOn(resultService, 'getResults').mockReturnValue(of(new HttpResponse<Result[]>({ body: [result] })));
 
         component.refresh();
 
-        expect(resultServiceStub).to.have.been.calledOnceWith(component.exercise);
-        expect(component.results).to.have.same.members([result]);
-        expect(component.isLoading).to.be.false;
+        expect(resultServiceStub).toHaveBeenCalledTimes(1);
+        expect(resultServiceStub).toHaveBeenCalledWith(component.exercise);
+        expect(component.results).toContainValues([result]);
```
Comment (b-fein) [inline src/test/javascript/spec/component/exercises/shared/exercise-scores.component.spec.ts:283]:
```suggestion
        expect(component.results).toEqual([result]);
```
Using the basic Jest function should yield the same result here.

**Review by b-fein — DISMISSED** [review]
Looks good! Added a few small inline comments.

File: src/test/javascript/spec/component/exercises/shared/exercise-scores.component.spec.ts (hunk 1 above)
Comment (ole-ve) [reply src/test/javascript/spec/component/exercises/shared/exercise-scores.component.spec.ts]: Thanks for feedback, changed in 06c22c4.

File: src/test/javascript/spec/component/exercises/shared/exercise-scores.component.spec.ts [hunk 3]
```diff
@@ -1,24 +1,16 @@
...
-import { MockComponent, MockDirective, MockPipe } from 'ng-mocks';
-import { TranslateService } from '@ngx-translate/core';
-import * as sinon from 'sinon';
-import sinonChai from 'sinon-chai';
-import * as chai from 'chai';
+import { ActivatedRoute } from '@angular/router';
+import { MockComponent, MockDirective, MockPipe, MockModule } from 'ng-mocks';
 import { of, Subscription } from 'rxjs';
-import { FormsModule } from '@angular/forms';
+import { NgModel, NgSelectOption } from '@angular/forms';
```
Comment (b-fein) [inline src/test/javascript/spec/component/exercises/shared/exercise-scores.component.spec.ts:13]:
```suggestion
import { NgModel } from '@angular/forms';
```
Just `NgModel` should be enough for this component, as there are no `select` drop-down menus in the HTML.

File: src/test/javascript/spec/component/exercises/shared/exercise-scores.component.spec.ts [hunk 4]
```diff
@@ -106,14 +96,15 @@ describe('Exercise Scores Component', () => {
                 MockHasAnyAuthorityDirective,
                 MockPipe(ArtemisTranslatePipe),
                 MockPipe(ArtemisDatePipe),
+                MockDirective(NgModel),
+                MockDirective(NgSelectOption),
```
Comment (b-fein) [inline src/test/javascript/spec/component/exercises/shared/exercise-scores.component.spec.ts:100]:
```suggestion
```

**Review by b-fein — COMMENTED** [review]
Thanks for implementing the feedback, just one last comment remaining.

File: src/test/javascript/spec/component/exercises/shared/exercise-scores.component.spec.ts (hunk 4 above)
Comment (ole-ve) [reply src/test/javascript/spec/component/exercises/shared/exercise-scores.component.spec.ts]: Changed, works also fine without.

**Review by b-fein — DISMISSED** [review]

File: src/test/javascript/spec/component/exercises/shared/exercise-scores.component.spec.ts [hunk 5]
```diff
@@ -86,13 +78,11 @@ describe('Exercise Scores Component', () => {
...
     result.assessmentType = AssessmentType.MANUAL;
 
-    const router = new MockRouter();
-
     const route = { data: of({ courseId: 1 }), children: [] } as any as ActivatedRoute;
 
     beforeEach(() => {
         TestBed.configureTestingModule({
-            imports: [ArtemisTestModule, RouterTestingModule.withRoutes([]), TranslateTestingModule, FormsModule, NgxDatatableModule],
+            imports: [ArtemisTestModule, RouterTestingModule.withRoutes([]), MockModule(NgxDatatableModule)],
```
Comment (MDK5) [inline src/test/javascript/spec/component/exercises/shared/exercise-scores.component.spec.ts:85]:
```suggestion
            imports: [ArtemisTestModule, MockModule(NgxDatatableModule)],
```
Most times this is not needed

File: src/test/javascript/spec/component/exercises/shared/exercise-scores.component.spec.ts [hunk 6]
```diff
@@ -106,14 +96,14 @@ describe('Exercise Scores Component', () => {
...
                 MockPipe(ArtemisDatePipe),
+                MockDirective(NgModel),
             ],
             providers: [
-                { provide: TranslateService, useClass: MockTranslateService },
                 { provide: ActivatedRoute, useValue: route },
-                { provide: AlertService, useClass: MockAlertService },
-                { provide: Router, useValue: router },
                 { provide: LocalStorageService, useClass: MockSyncStorage },
                 { provide: SessionStorageService, useClass: MockSyncStorage },
```
Comment (MDK5) [inline src/test/javascript/spec/component/exercises/shared/exercise-scores.component.spec.ts:104]: You could try to instead mock the Profile-, CourseManagement and ProgrammingSubmissionService. Or did you try that already and it did not work?

**Review by MDK5 — CHANGES_REQUESTED** [review]
Changes mostly look good to me, but I think the imports can be further improved.

File: src/test/javascript/spec/component/exercises/shared/exercise-scores.component.spec.ts (hunk 5 above)
Comment (ole-ve) [reply src/test/javascript/spec/component/exercises/shared/exercise-scores.component.spec.ts:85]: I tried without but tests were failing. In this case, it seems to be needed.

File: src/test/javascript/spec/component/exercises/shared/exercise-scores.component.spec.ts (hunk 6 above)
Comment (ole-ve) [reply src/test/javascript/spec/component/exercises/shared/exercise-scores.component.spec.ts]: Changed in 28b2984, works fine.

**Review by b-fein — APPROVED** [review]

**Review by willergeorg — APPROVED** [review]
Code changes lgtm

**Review by cremertim — APPROVED** [review]

**Review by MDK5 — APPROVED** [review]
Change requests got integrated, thanks.

**Pull request merged**